contracts/InterchainTokenFactory.sol

```diff
@@ -184,9 +268,25 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
 
         if (minter != address(0)) {
             if (!token.isMinter(minter)) revert NotMinter(minter);
+            // Sanity check to prevent accidental use of the current ITS address as the destination minter
             if (minter == address(interchainTokenService)) revert InvalidMinter(minter);
 
-            minter_ = minter.toBytes();
+            if (destinationMinter.length > 0) {
```

Comment:
So if I understand this correctly the minter can approve someone else (the 'deployer') to deploy a remote token to a specific chain with a specific minter. If this is true, why don't they just deploy it themselves?